src/Components/Web.JS/src/Rendering/WebRendererInteropMethods.ts

```diff
@@ -5,32 +5,32 @@ import { DotNet } from '@microsoft/dotnet-js-interop';
 import { EventDescriptor } from './Events/EventDelegator';
 import { enableJSRootComponents, JSComponentParametersByIdentifier, JSComponentIdentifiersByInitializer } from './JSRootComponents';
 
-const interopMethodsByRenderer = new Map<number, DotNet.DotNetObject>();
+const interopMethodsByRendererId: DotNet.DotNetObject[] = [];
 
 let resolveRendererAttached : () => void;
 
 export const rendererAttached = new Promise<void>((resolve) => {
   resolveRendererAttached = resolve;
 });
 
+// TODO: Consider allowing JS initializers to run multiple times.
+// We could add a "platform" argument so the initializer can handle each
+// platform's initialization separately, if needed.
 export function attachWebRendererInterop(
-  rendererId: number,
   interopMethods: DotNet.DotNetObject,
   jsComponentParameters: JSComponentParametersByIdentifier,
   jsComponentInitializers: JSComponentIdentifiersByInitializer,
-): void {
-  if (interopMethodsByRenderer.has(rendererId)) {
-    throw new Error(`Interop methods are already registered for renderer ${rendererId}`);
-  }
-
-  interopMethodsByRenderer.set(rendererId, interopMethods);
+): number {
```

Comment:
Before this PR, this method accepted a `rendererId` that was allocated on the .NET side. I've flipped this so that the `rendererId` is allocated in the browser and returned to .NET. That way, you don't have different .NET runtimes both starting at `0` and overwriting each other.